docs/user/analytic_queries.rst

```diff
@@ -0,0 +1,245 @@
+QueryHints.QUERY_INDEX ``GeoMesaFeatureIndex`` index name, or name:version
+====================== ======================= ===========================
+
+Java
+^^^^
+
+.. code-block:: java
+
+    import org.locationtech.geomesa.accumulo.index.z2.Z2Index$;
+    import org.locationtech.geomesa.index.conf.QueryHints;
+
+    query.getHints().put(QueryHints.QUERY_INDEX(), Z2Index$.MODULE$);
+
+Scala
+^^^^^
+
+.. code-block:: scala
+
+    import org.locationtech.geomesa.accumulo.index.z2.Z2Index
+    import org.locationtech.geomesa.index.conf.QueryHints
+
+    query.getHints.put(QueryHints.QUERY_INDEX, Z2Index)
+
+GeoServer
+^^^^^^^^^
+
+.. code-block:: none
+
+    ...&viewparams=QUERY_INDEX:z2;
+
+Query Planning
+--------------
+
+As explained above, GeoMesa uses cost-based query planning to determine the best index for a given query.
+If cost-based query planning is not working as desired, the legacy heuristic-based query
+planning can be used as a fall-back. ``Stats`` uses cost-based planning; ``Index`` uses heuristic-based planning.
+
+========================== ================== ======================
+Key                        Type               GeoServer Conversion
+========================== ================== ======================
+QueryHints.COST_EVALUATION ``CostEvaluation`` ``stats`` or ``index``
+========================== ================== ======================
+
+Java
+^^^^
+
+.. code-block:: java
+
+    import org.locationtech.geomesa.index.api.QueryPlanner.CostEvaluation;
+    import org.locationtech.geomesa.index.conf.QueryHints;
+
+    query.getHints().put(QueryHints.COST_EVALUATION(), CostEvaluation.Index());
+
+Scala
+^^^^^
+
+.. code-block:: scala
+
+    import org.locationtech.geomesa.index.api.QueryPlanner.CostEvaluation
+    import org.locationtech.geomesa.index.conf.QueryHints
+
+    query.getHints.put(QueryHints.COST_EVALUATION, CostEvaluation.Index)
+
+GeoServer
+^^^^^^^^^
+
+.. code-block:: none
+
+    ...&viewparams=COST_EVALUATION:index;
+
+Feature Sampling
+----------------
+
+Instead of returning all features for a query, GeoMesa can use statistical sampling to return a certain
+percentage of results. This can be useful when rendering maps, or when there are too many features to
+be meaningful.
+
+Features can either be sampled absolutely, or sampled by a certain attribute. For example, given a series of
+points in a track, you may wish to sample by the track identifier so that no tracks are completely sampled out.
+
+.. note::
+
+    Currently this section applies only to the Accumulo Data Store.
+
+========================== ================================== ====================
+Key                        Type                               GeoServer Conversion
+========================== ================================== ====================
+QueryHints.SAMPLING        Float
+QueryHints.SAMPLE_BY       String - attribute name (optional)
```

Comment:
Did you mean to leave the GeoServer Conversion column blank? Might be a good idea to put a sample value (or describe what values are valid). For example, you put 0.1 as a value in the Java/Scala examples below. Is that a fractional value or a percentage? (I.e. are 10% or 0.1% of the features sampled?)